app/collins/controllers/actions/ipaddress/DeleteAction.scala

```diff
@@ -22,27 +23,36 @@ case class DeleteAction(
   assetTag: String,
   spec: SecuritySpecification,
   handler: SecureController
-) extends SecureAction(spec, handler) with AssetAction with ParamValidation {
+) extends SecureAction(spec, handler) with AssetAction with ParamValidation with AddressActionHelper {
 
-  case class ActionDataHolder(asset: Asset, pool: Option[String]) extends RequestDataHolder
+  case class ActionDataHolder(asset: Asset, pool: Option[String], address: Option[String]) extends RequestDataHolder
 
-  val dataForm = Form(
-    "pool" -> validatedOptionalText(1)
-  )
+  val dataForm = Form(tuple(
+    "pool" -> validatedOptionalText(1),
+    "address" -> validatedOptionalText(1)
+  ))
 
   override def validate(): Validation = withValidAsset(assetTag) { asset =>
-    val pool: Option[String] = dataForm.bindFromRequest()(request).fold(
-      err => None,
-      str => str
-    )
-    Right(ActionDataHolder(asset, pool))
+    val (pool, address): Tuple2[Option[String], Option[String]] = dataForm.bindFromRequest()(request).get
+    address match {
+      case Some(ip: String) => withValidAddress(ip) { validAddress => 
+        Right(ActionDataHolder(asset, pool, address))
+      }
+      case None => Right(ActionDataHolder(asset, pool, None))
+    }
```

Comment:
Sure.  I didn't add any tests cause there weren't any for any of these classes already.  I'll take a stab at adding some tests.